modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala

```diff
@@ -49,6 +40,15 @@ class HttpResponseCodeSchemaVisitor()
     case _ => NoResponseCode
   }
 
+
+  override def enumeration[E](shapeId: ShapeId, hints: Hints, values: List[EnumValue[E]], total: E => EnumValue[E]): ResponseCodeExtractor[E] =
+    if (hints.has[smithy.api.HttpResponseCode] && hints.has[IntEnum]) {
```

Comment:
**thought:** maybe we don't need the second part of the conditional, as it should already be covered by smithy's own validation?

Reply:
yeah, I'm never sure either :) it's fine to keep it as-is